### PR TITLE
Chore/allow share post on twitter

### PR DESCRIPTION
**Pull request by vicmaster**
## Quick Info

Allow users to share post `TIL's` to Twitter.

### Boy Scout Rule

In the beginning, my purpose was only to bring back the tweet button, but our `RSpec` suite was super broken, so I decided to fix some specs, not all of them. And I do also marked some specs as pending.

```
Finished in 2.2 seconds (files took 7.74 seconds to load)
89 examples, 0 failures, 16 pending
``` 

## What does this change?

Because sharing is caring

## Why are you changing that?

Because I think it's going to help us to improve post sharing

## Migrations?

No 👎 

## How do you manually test this?

Yes, but only in my local environment

## Screenshots

Screen Shot 2021-02-23 at 3 51 49 AM

### CODE REVIEW

**Review by fedgut — APPROVED** [review]
LGTM!

File: config/routes.rb [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-
   get '/auth/:provider/callback' => 'sessions#create'
   delete '/signout' => 'sessions#destroy', :as => :signout
   get '/auth/failure' => 'sessions#failure'
+  resource :session, only: [:create, :destroy]
```
Comment (softr8) [inline config/routes.rb:7]: ? why is this needed? we're not using devise anymore

Comment (vicmaster) [reply config/routes.rb:7]: I know, but when I was trying to Fix the SessionController spec `post` request was broken with a UrlGenerationError like this:

```
 ActionController::UrlGenerationError:
       No route matches {:action=>"create", :controller=>"sessions"}
```

that's why I added on `routes` not sure if there is another approach to this

File: spec/controllers/sessions_controller_spec.rb [hunk 2]
```diff
@@ -3,37 +3,78 @@
...
+
     context 'with oauth' do
       it 'signs in existing developer by email' do
-        developer = FactoryBot.create :developer, email: 'jake@example.com'
-        request.env['omniauth.auth'] = {
-            'info' => {
-                'email' => 'jake@example.com'
-            }
-        }
+        developer = FactoryBot.create :developer, email: 'jake@magmalabs.io'
```
Comment (softr8) [inline spec/controllers/sessions_controller_spec.rb:17]: what about re-using the `flow@magmalabs.io` it was already mocked for omniauth

Comment (softr8) [reply spec/controllers/sessions_controller_spec.rb:17]: Nevermind, it's the new user

File: spec/controllers/sessions_controller_spec.rb [hunk 3]
```diff
@@ -3,37 +3,78 @@
...
+          }
+        )
+      end
 
+      it 'a new developer by email' do
         expect do
           post :create
           expect(response).to redirect_to root_path
         end.to change(Developer, :count).by(1)
         developer = Developer.last
```
Comment (softr8) [inline spec/controllers/sessions_controller_spec.rb]: I tend not to trust in first/last/etc speaking about Postgresql and sql server, they return the rows as soon as they read them from disk, but that does not ensure they are sorted, it depends on internal disk availability

File: spec/models/post_spec.rb [hunk 4]
```diff
@@ -224,7 +150,7 @@
   describe '#publish' do
     it 'sets the post to published = true' do
       post.publish
-      expect(post.published_at).to be
+      expect(post.published_at).not_to be_nil
```
Comment (softr8) [inline spec/models/post_spec.rb]: no need to changue, but IMO positive assertions are easier to read: `to be_present`

File: app/models/social_messaging/twitter_status.rb [hunk 5]
```diff
@@ -11,11 +11,11 @@ def initialize(post)
     def post_to_twitter
       return if post.draft? || post.tweeted
 
-      if ENV['update_twitter_with_post'] == 'true'
-        TwitterClient.update(status)
-        post.tweeted = true
-        post.save
-      end
+      return unless ENV['update_twitter_with_post'] == 'true'
```
Comment (softr8) [inline app/models/social_messaging/twitter_status.rb]: it's a bit hard to understand easily what's trying to do, maybe:
```ruby
return if ENV.fetch('SKIP_TWITTER_POSTING', false)
```

File: app/models/social_messaging/twitter_status.rb [hunk 6]
```diff
@@ -11,11 +11,11 @@ def initialize(post)
...
+
+      TwitterClient.update(status)
+      post.tweeted = true
```
Comment (softr8) [inline app/models/social_messaging/twitter_status.rb]:
```ruby
post.update(tweeted: true)
```

File: app/models/social_messaging/twitter_status.rb [hunk 7]
```diff
@@ -32,13 +32,9 @@ def category
...
     end
 
-    def host
-      ENV.fetch('HOST')
-    end
-
     def status
       "#{title} #{Rails.application.routes.url_helpers.post_url(titled_slug: post.to_param,
-                                                                host: host)} via @#{name} #til ##{category}"
+                                                                host: 'til.magmalabs.io')} via @#{name} #til ##{category}"
```
Comment (softr8) [inline app/models/social_messaging/twitter_status.rb]: then why we need `ENV['HOST']`? if we hardcode til.magmalabs.io :p

File: spec/models/social_messaging/twitter_status_spec.rb [hunk 8]
```diff
@@ -22,7 +22,7 @@
 
   describe '#status' do
     it 'returns a Twitter status' do
-      host = 'www.magmalabs.io'
+      host = 'til.magmalabs.io'
```
Comment (softr8) [inline spec/models/social_messaging/twitter_status_spec.rb:25]: same above about ENV['HOST']

File: spec/helpers/post_helper_spec.rb [hunk 9]
```diff
@@ -4,11 +4,11 @@
 
 describe PostHelper do
   describe '#tweet_link' do
-    context 'returns a link to twitter' do
+    context 'when returns a link to twitter' do
       specify 'with appropriate data' do
```
Comment (softr8) [inline spec/helpers/post_helper_spec.rb]: specify is coming from older days of rspec, maybe it's worth refactoring it to `it`

File: app/models/social_messaging/twitter_status.rb (hunk 7 above)
Comment (vicmaster) [reply app/models/social_messaging/twitter_status.rb]: If you see closely, I removed that method ;)

File: spec/models/social_messaging/twitter_status_spec.rb (hunk 8 above)
Comment (vicmaster) [reply spec/models/social_messaging/twitter_status_spec.rb:25]: same answer as above

File: app/models/social_messaging/twitter_status.rb (hunk 5 above)
Comment (vicmaster) [reply app/models/social_messaging/twitter_status.rb]: I was just following rubocop on that approach

File: app/models/social_messaging/twitter_status.rb (hunk 6 above)
Comment (vicmaster) [reply app/models/social_messaging/twitter_status.rb]: Nice

**Review by softr8 — APPROVED** [review]
